gateway/api/runbooks/rules.go

```diff
@@ -0,0 +1,209 @@
+// @Description  Get a single Runbook Rule by ID
+// @Tags         Runbooks
+// @Produce      json
+// @Param        id  path      string  true  "Runbook Rule ID"
+// @Success      200  {object}  openapi.RunbookRule
+// @Failure      404,422,500  {object} openapi.HTTPError
+// @Router       /runbooks/rules/{id} [get]
+func GetRunbookRule(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+
+	ruleID := c.Param("id")
+
+	rule, err := models.GetRunbookRuleByID(models.DB, ctx.GetOrgID(), ruleID)
+	if err != nil {
+		log.Infof("failed fetching runbook rule, err=%v", err)
+		c.JSON(http.StatusNotFound, gin.H{"message": "runbook rule not found"})
+		return
+	}
+
+	responseRule := buildRunbookResponseFromModel(rule)
+	c.JSON(http.StatusOK, responseRule)
+}
+
+// CreateRunbookRule
+//
+// @Summary      Create Runbook Rule
+// @Description  Create a new Runbook Rule
+// @Tags         Runbooks
+// @Accept       json
+// @Produce      json
+// @Param        rule  body     openapi.RunbookRule  true  "Runbook Rule"
+// @Success      201  {object}  openapi.RunbookRule
+// @Failure      400,422,500  {object} openapi.HTTPError
+// @Router       /runbooks/rules [post]
+func CreateRunbookRule(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+
+	var reqRule openapi.RunbookRule
+	if err := c.ShouldBindJSON(&reqRule); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "invalid request body"})
+		return
+	}
+
+	rule := buildRunbookRuleModelFromRequest(&reqRule)
+	rule.OrgID = ctx.GetOrgID()
+
+	if err := models.UpsertRunbookRule(models.DB, &rule); err != nil {
+		log.Errorf("failed creating runbook rule, reason=%v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed creating runbook rule"})
+		return
+	}
+
+	ruleRes := buildRunbookResponseFromModel(&rule)
+
+	c.JSON(http.StatusCreated, ruleRes)
+}
+
+// UpdateRunbookRule
+//
+// @Summary      Update Runbook Rule
+// @Description  Update an existing Runbook Rule by ID
+// @Tags         Runbooks
+// @Accept       json
+// @Produce      json
+// @Param        id    path      string              true  "Runbook Rule ID"
+// @Param        rule  body      openapi.RunbookRule  true  "Runbook Rule"
+// @Success      200  {object}   openapi.RunbookRule
+// @Failure      400,404,422,500  {object} openapi.HTTPError
+// @Router       /runbooks/rules/{id} [put]
+func UpdateRunbookRule(c *gin.Context) {
+	ctx := storagev2.ParseContext(c)
+
+	ruleID := c.Param("id")
+
+	var reqRule openapi.RunbookRule
+	if err := c.ShouldBindJSON(&reqRule); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "invalid request body"})
+		return
+	}
+
+	rule := buildRunbookRuleModelFromRequest(&reqRule)
+	rule.ID = ruleID
+	rule.OrgID = ctx.GetOrgID()
+
+	if err := models.UpsertRunbookRule(models.DB, &rule); err != nil {
+		log.Errorf("failed updating runbook rule, reason=%v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed updating runbook rule"})
```

Comment:
Just return the error to the client, it will help us to identify issues in the client without having to inspect gateway logs.